### PR TITLE
chore: update cni to v1.4.16

### PR DESCRIPTION
**Pull request by AbelHu**
**Reason for Change**:

https://github.com/Azure/azure-container-networking/releases/tag/v1.4.16
82f4b39 feat: [CNI] Enable DSR policy for windows singletenancy (AKS) by default (#1116)

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by ShiqianTao — APPROVED** [review]

File: vhd/packer/configure-windows-vhd.ps1 [hunk 1]
```diff
@@ -128,7 +128,8 @@ function Get-FilesToCacheOnVHD {
         "c:\akse-cache\win-vnet-cni\" = @(
             "https://kubernetesartifacts.azureedge.net/azure-cni/v1.4.12/binaries/azure-vnet-cni-singletenancy-windows-amd64-v1.4.12.zip",
```
Comment (tamilmani1989) [inline vhd/packer/configure-windows-vhd.ps1:129]: remove 1.4.12 release? as aks-e used to keep only last 3 release

File: vhd/packer/install-dependencies.sh [hunk 2]
```diff
@@ -96,6 +96,7 @@ installEtcd "docker"
 echo "  - etcd v${ETCD_VERSION}" >> ${VHD_LOGS_FILEPATH}
 
 VNET_CNI_VERSIONS="
+1.4.16
 1.4.14
 1.4.13
 1.4.12
```
Comment (tamilmani1989) [inline vhd/packer/install-dependencies.sh:102]: same as above..remove v1.4.12?

**Comment by jsturtevant** [comment]
do we need changes to the CNI configuration to enable DSR with this new package?

**Review by jackfrancis — APPROVED** [review]
/lgtm

**Comment by AbelHu** [comment]
> do we need changes to the CNI configuration to enable DSR with this new package?

@jsturtevant WinDSR support is not added in aks-engine. Will add the settings in azure cni conflist when adding WinDSR support.

**Comment by tamilmani1989** [comment]
> > do we need changes to the CNI configuration to enable DSR with this new package?
> 
> @jsturtevant WinDSR support is not added in aks-engine. Will add the settings in azure cni conflist when adding WinDSR support.

There is a way to enable DSR in aks-engine with cluster definition. If DSR is enabled, then we need to update cni conflist to pass that setting.

**Comment by AbelHu** [comment]
@tamilmani1989 aks-engine has not supported WinDSR. I am testing loopbackDSR setting in AKS

**Comment by tamilmani1989** [comment]
> @tamilmani1989 aks-engine has not supported WinDSR. I am testing loopbackDSR setting in AKS

@AbelHu i tested it in aks-engine cluster with above feature enabled and it worked

**Comment by AbelHu** [comment]
@tamilmani1989 I tested v1.4.16 with this change https://github.com/Azure/AgentBaker/commit/cf9784b65a3399a15c4d212c74542ddba4117504. It seems like that the fix works.

```
~/croot/temp/testwinan$ kubectl exec -it deployment/win-webserver -- curl.exe -v --max-time 10 http://localhost:80 -H "Connection: close"
* Rebuilt URL to: http://localhost:80/
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 80 (#0)
> GET / HTTP/1.1
> Host: localhost
> User-Agent: curl/7.55.1
> Accept: */*
> Connection: close
>
< HTTP/1.1 200 OK
< Content-Length: 95
< Server: Microsoft-HTTPAPI/2.0
< Date: Fri, 03 Dec 2021 04:47:59 GMT
< Connection: close
<
<html><body><H1>Windows Container Web Server</H1><p>IP 10.240.0.12 callerCount 1 </body></html>* Closing connection 0

~/croot/temp/testwinan$ kubectl exec -it deployment/win-webserver -- curl.exe -v --max-time 10 http://win-webserver:80 -H "Connection: close"
* Rebuilt URL to: http://win-webserver:80/
*   Trying 10.0.10.173...
* TCP_NODELAY set
* Connected to win-webserver (10.0.10.173) port 80 (#0)
> GET / HTTP/1.1
> Host: win-webserver
> User-Agent: curl/7.55.1
> Accept: */*
> Connection: close
>
< HTTP/1.1 200 OK
< Content-Length: 127
< Server: Microsoft-HTTPAPI/2.0
< Date: Fri, 03 Dec 2021 04:48:07 GMT
< Connection: close
<
<html><body><H1>Windows Container Web Server</H1><p>IP 10.240.0.12 callerCount 1 <p>IP 10.240.0.12 callerCount 1 </body></html>* Closing connection 0
```

**Pull request merged**